cmd/influxd/inspect/tsm_verify.go

```diff
@@ -165,68 +122,77 @@ func (v *verifyChecksums) Run(w io.Writer, dataPath string) error {
 			if err != nil {
 				v.totalErrors++
 				fileErrors++
-				fmt.Fprintf(w, "%s: could not get checksum for key %v block %d due to error: %q\n", f, key, count, err)
+				if verbose {
+					cmd.PrintErrf("%s: could not get checksum for key %v block %d due to error: %q\n", f, key, count, err)
+				}
 			} else if expected := crc32.ChecksumIEEE(buf); checksum != expected {
 				v.totalErrors++
 				fileErrors++
-				fmt.Fprintf(w, "%s: got %d but expected %d for key %v, block %d\n", f, checksum, expected, key, count)
+				if verbose {
+					cmd.PrintErrf("%s: got %d but expected %d for key %v, block %d\n", f, checksum, expected, key, count)
+				}
 			}
 			count++
 		}
-		if fileErrors == 0 {
-			fmt.Fprintf(w, "%s: healthy\n", f)
+		if fileErrors == 0 && verbose {
+			cmd.PrintErrf("%s: healthy\n", f)
 		}
 		reader.Close()
 	}
 
-	fmt.Fprintf(w, "Broken Blocks: %d / %d, in %vs\n", v.totalErrors, v.total, v.Elapsed().Seconds())
+	cmd.PrintErrf("Broken Blocks: %d / %d, in %vs\n", v.totalErrors, v.total, v.Elapsed().Seconds())
 
 	return v.err
 }
 
-type verifyUTF8 struct {
-	verifyTSM
-	totalErrors int
-	total       int
-}
+func (v *verifyTSM) loadFiles(dataPath string) error {
+	err := filepath.Walk(dataPath, func(path string, f os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Ext(path) == "."+tsm1.TSMFileExtension {
+			v.files = append(v.files, path)
+		}
+		return nil
+	})
 
-func (v *verifyUTF8) Run(w io.Writer, dataPath string) error {
-	if err := v.loadFiles(dataPath); err != nil {
-		return err
+	if err != nil {
+		return errors.Wrap(err, "could not load storage files (use -dir for custom storage root)")
 	}
 
-	v.Start()
+	return nil
+}
 
-	for v.Next() {
-		f, reader := v.TSMReader()
-		if reader == nil {
-			break
-		}
+func (v *verifyTSM) Next() bool {
+	if len(v.files) == 0 {
+		return false
+	}
 
-		n := reader.KeyCount()
-		fileErrors := 0
-		v.total += n
-		for i := 0; i < n; i++ {
-			key, _ := reader.KeyAt(i)
-			if !utf8.Valid(key) {
-				v.totalErrors++
-				fileErrors++
-				fmt.Fprintf(w, "%s: key #%d is not valid UTF-8\n", f, i)
-			}
-		}
-		if fileErrors == 0 {
-			fmt.Fprintf(w, "%s: healthy\n", f)
-		}
+	v.f, v.files = v.files[0], v.files[1:]
+	return true
+}
+
+func (v *verifyTSM) TSMReader() (string, *tsm1.TSMReader) {
+	file, err := os.OpenFile(v.f, os.O_RDONLY, 0600)
+	if err != nil {
+		v.err = err
```

Comment:
Setting the `err` instead of returning it isn't the pattern I'd usually expect, could we have it as a return value for this function instead? What's the purpose of `v.err`?